### PR TITLE
Support wait-for-data timeout

### PR DESCRIPTION
**Pull request by mtojek**
## Checklist

- [x] I have added test packages to [`test/packages`](https://github.com/elastic/package-spec/tree/master/test/packages) that prove my change is effective.
- [ ] I have added an entry in [`versions/N/changelog.yml`](https://github.com/elastic/package-spec/blob/master/versions/1/changelog.yml).

## Related issues

- https://github.com/elastic/elastic-package/issues/540

### CODE REVIEW

File: versions/1/data_stream/_dev/test/system/config.spec.yml [hunk 1]
```diff
@@ -8,3 +8,7 @@ spec:
   properties:
     skip:
       $ref: "./../skip.spec.yml#/definitions/skip"
+    wait_for_data_timeout:
+      description: Short explanation for why test has been skipped.
```
Comment (jsoriano) [inline versions/1/data_stream/_dev/test/system/config.spec.yml:12]: Umm, this description is unrelated? :thinking:

File: versions/1/data_stream/_dev/test/system/config.spec.yml [hunk 2]
```diff
@@ -8,3 +8,7 @@ spec:
...
+      type: string
```
Comment (jsoriano) [inline versions/1/data_stream/_dev/test/system/config.spec.yml:13]: Q: Are types defined somewhere? Would it be possible to define a duration type for this strings?

File: versions/1/data_stream/_dev/test/system/config.spec.yml (hunk 1 above)
Comment (mtojek) [reply versions/1/data_stream/_dev/test/system/config.spec.yml]: My bad, copied this structure from the "skip" definition.

**Review by jsoriano — APPROVED** [review]
